File: java/src/privileges/Approve.java
package privileges;

import decorator.PrivilegeDecorator;
import decorator.User;

public class Approve extends PrivilegeDecorator{
	User user;
	
	public Approve(User user) {
		this.user = user;
	}

	@Override
	public String getPrivileges() {
		return user.getPrivileges() + " approve";
	}

	@Override
	public String getName() {
		return user.getName();
	}
}

File: csharp/decorator_practice/decorator_practice/main/privileges/Approve.cs
using decorator_practice.main.decorator;
using System;
using System.Collections.Generic;
using System.Text;

namespace decorator_practice.main.privileges
{
    public class Approve : PrivilegeDecorator{
        /**
	     * User.
	     */
        private User user;

        /**
	     * @param u constructor.
	     */
        public Approve(User u)
        {
            this.user = u;
        }

        /**
	     * @return approve privilege.
	     */
        override
        public String getPrivileges()
        {
            return user.getPrivileges() + " approve";
        }

        /**
	     * @return name.
	     */
        override
        public String getName()
        {
            return user.getName();
        }
    }

}
